kamailio/sip_trace-role.cfg

```diff
@@ -0,0 +1,17 @@
+####### Siptrace module  ##########
+
+loadmodule "siptrace.so"
+modparam("siptrace", "duplicate_uri", "SIP_TRACE_URI")
+modparam("siptrace", "hep_mode_on", 1)
+modparam("siptrace", "hep_version", 3)
+modparam("siptrace", "hep_capture_id", HEP_CAPTURE_ID)
+modparam("siptrace", "trace_to_database", 0)
+## `trace_flag` value must be equal to FLAG_SIP_TRACE value at head of "default.cfg" file
+modparam("siptrace", "trace_flag", 8)
+modparam("siptrace", "trace_on", 1)
+
+route[SEND_SIP_TRACE]
+{
+    setflag(FLAG_SIP_TRACE);
+    sip_trace();
+}
```

Comment:
@sergey-safarov here's some additions that i would like to be added, comments appreciated. these options allow  for the role to be enabled but controls what parts it should do and can be changed at runtime  (we may also have homer capture on freeswitch).

```
kazoo.trace = KZ_TRACE descr "activates global tracing"
kazoo.trace_external = KZ_TRACE_INTERNAL descr "activates tracing from external sources"
kazoo.trace_internal = KZ_TRACE_EXTERNAL descr "activates tracing from internal sources"

kazoo.trace_internal_incoming = KZ_TRACE_INTERNAL_INCOMING descr "traces the original request as received from internal sources"
kazoo.trace_internal_outgoing = KZ_TRACE_INTERNAL_OUTGOING descr "traces the outgoing request to external sources after possible modification"
kazoo.trace_external_incoming = KZ_TRACE_EXTERNAL_INCOMING descr "traces the original request as received from external sources"
kazoo.trace_external_outgoing = KZ_TRACE_EXTERNAL_OUTGOING descr "traces the outgoing request to internal sources after possible modification"

route[SIP_TRACE_INTERNAL]
{
     if(@cfg_get.kazoo.trace_internal == 0) {
          return;
     }
    if(@cfg_get.kazoo.trace_internal_incoming  == 1) {
          sip_trace();
   }
    if(@cfg_get.kazoo.trace_internal_outgoing == 1) {
          setflag(FLAG_SIP_TRACE);
   }
}

route[SIP_TRACE_EXTERNAL]
{
.... same as above but proper selectors
}

route[SEND_SIP_TRACE]
{
     if(@cfg_get.kazoo.trace  == 0) {
          return;
     }

    if (isflagset(FLAG_INTERNALLY_SOURCED)) {
         route(SIP_TRACE_INTERNAL);
     } else {
         route(SIP_TRACE_EXTERNAL);
     }
}

```
the `KZ_TRACE` and friends should be declared in defs.cfg as #!ifndef .. #!define .. so they can be overridden in local.cfg. the default values could be something like the below (suggestions for better default welcomed)

```
#!ifndef KZ_TRACE
#!define KZ_TRACE 0
#!endif

##note : the above can probably be simplified with  a single line
## needs testing and probably apply to other integer defines in defs.cfg
`#!trydef KZ_TRACE 0`

... ommitting the preprocessor macro for simplicity
KZ_TRACE_INTERNAL 1
KZ_TRACE_EXTERNAL 1
KZ_TRACE_INTERNAL_INCOMING 1
KZ_TRACE_INTERNAL_OUTGOING 1
KZ_TRACE_EXTERNAL_INCOMING 1
KZ_TRACE_EXTERNAL_OUTGOING 1
```